scripts/checkpoint.py

```diff
@@ -0,0 +1,50 @@
+import os
+import glob
+
+# Examples:
+# sim_waveform_dirs =
+# ['/nesi/nobackup/nesi00213/RunFolder/Cybershake/v18p6/Runs/test/Kelly/BB/Cant1D_v3-midQ_OneRay_hfnp2mm+_rvf0p8_sd50_k0p045/Kelly_HYP20-29_S1434',
+# '/nesi/nobackup/nesi00213/RunFolder/Cybershake/v18p6/Runs/test/Kelly/BB/Cant1D_v3-midQ_OneRay_hfnp2mm+_rvf0p8_sd50_k0p045/Kelly_HYP29-29_S1524',
+# '/nesi/nobackup/nesi00213/RunFolder/Cybershake/v18p6/Runs/test/Kelly/BB/Cant1D_v3-midQ_OneRay_hfnp2mm+_rvf0p8_sd50_k0p045/Kelly_HYP07-29_S1304']
+# dire = '/nesi/nobackup/nesi00213/RunFolder/Cybershake/v18p6/Runs/test/Kelly/BB/Cant1D_v3-midQ_OneRay_hfnp2mm+_rvf0p8_sd50_k0p045/Kelly_HYP20-29_S1434'
+# output_sim_dir = /nesi/nobackup/nesi00213/RunFolder/Cybershake/v18p6/Runs/test/Kelly/BB/Cant1D_v3-midQ_OneRay_hfnp2mm+_rvf0p8_sd50_k0p045/Kelly_HYP20-29_S1434/../../../IM_calc/Kelly_HYP20-29_S1434/
+
+
+def checkpoint_sim_obs(waveform_dirs, relative_path_to_im_calc):
```

Comment:
Dont use relative path. 
you have the run_dir, sim_dir. fault_name, generate the full-path and enforce the rule.
example pseudo code: 
`IM_calc_dir = join(sim_dir,'IM_calc');
for hyp in hpy_list: 
do
     IM_hpy = join(IM_calc_dir,hyp);
`

Reply:
if no using relative path, function would be:
def checkpoint_sim_obs(run_dir, waveform_dirs):
    for directory in waveform_dirs[:]:
        dir_name = directory.split('/')[-1]
        if dir_name == 'IM_calc':
            waveform_dirs.remove(directory)
        else:
           fault_name = dir_name.split('_')[0]
            output_dir = os.path.join(run_dir, fault_name, 'IM_calc')
            if os.path.isdir(output_dir):  # if output dir exists
                sum_csv = glob.glob1(output_dir, '*.csv')
                meta = glob.glob1(output_dir, '*imcalc.info')
                # if sum_csv and meta are not empty lists('.csv' and '_imcalc.info' files present)
                # then we think im calc on the corresponding dir is completed and hence remove
                if sum_csv and meta:
                    waveform_dirs.remove(directory)
    return waveform_dirs